src/kerning.rs

```diff
@@ -0,0 +1,90 @@
+use std::collections::BTreeMap;
+
+use serde::ser::{SerializeMap, Serializer};
+use serde::Serialize;
+
+/// Kerning is a map of first half of a kerning pair (glyph name or group name) to the second half
+/// of a pair (glyph name or group name), which maps to the kerning value (high-level view:
+/// (first, second) => value). It's a BTreeMap because we need sorting for serialization.
+pub type Kerning = BTreeMap<String, BTreeMap<String, f32>>;
+
+/// KerningSerializer is a crutch to serialize kerning values as integers if they are
+/// integers rather than floats. This spares us having to use a wrapper type like
+/// IntegerOrFloat for kerning values.
```

Comment:
```suggestion
/// A helper for serializing kerning values.
///
/// KerningSerializer is a crutch to serialize kerning values as integers if they are
/// integers rather than floats. This spares us having to use a wrapper type like
/// `IntegerOrFloat` for kerning values.
```